src/librustdoc/html/render/mod.rs

```diff
@@ -2582,16 +2578,20 @@ fn item_trait(w: &mut Buffer, cx: &Context<'_>, it: &clean::Item, t: &clean::Tra
         write!(
             w,
             "{}{}{}trait {}{}{}",
-            it.visibility.print_with_space(cx.tcx(), it.def_id),
+            it.visibility.print_with_space(cx.tcx(), it.def_id, cx.cache()),
             t.unsafety.print_with_space(),
             if t.is_auto { "auto " } else { "" },
             it.name.as_ref().unwrap(),
-            t.generics.print(),
+            t.generics.print(cx.cache()),
             bounds
         );
 
         if !t.generics.where_predicates.is_empty() {
-            write!(w, "{}", WhereClause { gens: &t.generics, indent: 0, end_newline: true });
+            write!(
+                w,
+                "{}",
+                WhereClause { gens: &t.generics, indent: 0, end_newline: true }.print(cx.cache())
+            );
```

Comment:
Rustfmt has really bad taste here.
```suggestion
            let where_ = WhereClause { gens: &t.generics, indent: 0, end_newline: true };
            write!(w, "{}", where_.print(cx.cache()));
```